controllers/volumereplicationgroup_controller.go

```diff
@@ -1145,6 +1173,103 @@ func (v *VRGInstance) createVR(vrNamespacedName types.NamespacedName, state volr
 	return nil
 }
 
+// namespacedName applies to both VolumeReplication resource and pvc as of now.
+// This is because, VolumeReplication resource for a pvc that is created by the
+// VolumeReplicationGroup has the same name as pvc. But in future if it changes
+// functions to be changed would be processVRAsPrimary(), processVRAsSecondary()
+// to either receive pvc NamespacedName or pvc itself as an additional argument.
+func (v *VRGInstance) selectVolumeReplicationClass(namespacedName types.NamespacedName) (string, error) {
+	className := ""
+
+	if !v.vrcUpdated {
+		if err := v.updateReplicationClassList(); err != nil {
+			v.log.Error(err, "Failed to get VolumeReplicationClass list")
+
+			return className, fmt.Errorf("failed to get VolumeReplicationClass list")
+		}
+
+		v.vrcUpdated = true
+	}
+
+	if len(v.replClassList.Items) == 0 {
+		v.log.Info("No VolumeReplicationClass available")
+
+		return className, fmt.Errorf("no VolumeReplicationClass available")
+	}
+
+	storageClass, err := v.getStorageClass(namespacedName)
+	if err != nil {
+		v.log.Info(fmt.Sprintf("Failed to get the storageclass of pvc %s",
+			namespacedName))
+
+		return className, fmt.Errorf("failed to get the storageclass of pvc %s (%w)",
+			namespacedName, err)
+	}
+
+	for index := range v.replClassList.Items {
+		replicationClass := &v.replClassList.Items[index]
+		if storageClass.Provisioner != replicationClass.Spec.Provisioner {
+			continue
+		}
+
+		schedule, found := replicationClass.Spec.Parameters["schedule"]
+		if !found {
+			// schedule not present in parameters of this replicationClass.
+			continue
+		}
+
+		// ReplicationClass that matches both VRG schedule and pvc provisioner
+		if schedule == *v.instance.Spec.Schedule {
```

Comment:
As per [ceph-csi](https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/replicationcontrollerserver.go#L64-L68) this compare will fail as we use a cron spec and they are on an interval. (as stated above in the other comment).

We possibly can just move to the interval at present than use a cron spec for the same.

Reply:
ACK. Should the csi format of schedule be only made for VRG or do you think it needs to happen for higher layers such as DRPolicy etc? 

For simplicity I am fine with making everything interval based schedule as per CSI format.